server/tasks/createTask.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/* example of body input
+ {
+    "title": "task title",
+    "description": "task description test",
+    "reward": "coffe test bla",
+    "owner": 2,
+    "parent": 0
+}
+ */
+
+if ($_SERVER["REQUEST_METHOD"] !== "POST") {
+    header("HTTP/1.0 403 Forbidden");
+}
+
+include "../config.inc.php";
+include "../auth.inc.php";
+
+UserAuthentication();
+$response = [];
+
+function isValidJSON($str) : bool {
+    json_decode($str);
+    return json_last_error() == JSON_ERROR_NONE;
+}
+
+function FailRequest($error_msg) {
+    $response['success'] = false;
+    $response['error_msg'] = $error_msg;
+    echo json_encode($response, JSON_PRETTY_PRINT);
+    exit();
+}
+
+function ValidateMandatoryTaskParam($param) {
+    if (strlen($param) > 0) {
+        return htmlspecialchars($param);
+    }
+    else {
+        FailRequest("Bad request");
+    }
+}
+
+$json_params = file_get_contents("php://input");
```

Comment:
This is weird. If you want to go with implementing from scratch instead of using a framework, fine, but lets at least have by convention a function called "process_request" or something similar in each file. I.e. the user authentication and response definition happens, then some random functions, then some more code, etc. Trippy :)